src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2062,6 +2062,53 @@ Method* InstanceKlass::lookup_method_in_all_interfaces(Symbol* name,
   return NULL;
 }
 
+PrintClassClosure::PrintClassClosure(outputStream* st, bool verbose)
+  :_st(st), _verbose(verbose) {
+  ResourceMark rm;
+  _st->print("%-18s  ", "KlassAddr");
+  _st->print("%-4s  ", "Size");
+  _st->print("%-20s  ", "State");
+  _st->print("%-7s  ", "Flags");
+  _st->print("%-5s  ", "ClassName");
+  _st->cr();
+}
+
+void PrintClassClosure::do_klass(Klass* k)  {
+  ResourceMark rm;
+  // klass pointer
+  _st->print(INTPTR_FORMAT "  ", p2i(k));
+  // klass size
+  _st->print("%-4d  ", k->size());
```

Comment:
Should be `%4d` so that the numbers are aligned correctly.

Reply:
Current:
```
$./jcmd 83908 VM.classes|head -10
83908:
KlassAddr           Size  State                 Flags    ClassName  
0x0000000800df8400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8400  
0x0000000800df8000  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8000  
0x0000000800de4400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4400  
0x0000000800de4000  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4000  
0x0000000800dc8800  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800dc8800  
0x0000000800dc8400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800dc8400  
0x0000000800dc8000  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800dc8000  
0x0000000800db9800  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800db9800 
```
After using "%-4d":
```
$./jcmd 75481 VM.classes|head
75481:
KlassAddr           Size  State                 Flags    ClassName  
0x0000000800df8400    62  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8400  
0x0000000800df8000    62  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8000  
0x0000000800de4400    62  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4400  
0x0000000800de4000    62  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4000 
```
So we do not need to change this.

Reply:
You should change it to `%4d`. Otherwise, when the numbers are changed in the future (e.g., to 3 or 4 digits) they will be misaligned:

```
KlassAddr           Size  State                 Flags    ClassName  
0x0000000800df8400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8400  
0x0000000800df8000  123   fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8000  
0x0000000800de4400  4567  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4400  
```

Reply:
> You should change it to `%4d`. Otherwise, when the numbers are changed in the future (e.g., to 3 or 4 digits) they will be misaligned:
> 
> ```
> KlassAddr           Size  State                 Flags    ClassName  
> 0x0000000800df8400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8400  
> 0x0000000800df8000  123   fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8000  
> 0x0000000800de4400  4567  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4400  
> ```

This format looks pretty good to me, they are all aligned to left. If you still think it's more proper to have a format like this:
```
KlassAddr           Size  State                 Flags    ClassName  
0x0000000800df8400  62    fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8400  
0x0000000800df8000  123   fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800df8000  
0x0000000800de4400  4567  fully_initialized     W        java.lang.invoke.LambdaForm$DMH/0x0000000800de4400  
```
Then I'm glad to do so ;)

Reply:
Numbers should be aligned to the right. The following is what I want:

```
  62
 123
4567
```

Reply:
> Numbers should be aligned to the right. The following is what I want:
> 
> ```
>   62
>  123
> 4567
> ```

Done.